src/Subscriptions.tsx

```diff
@@ -0,0 +1,134 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import ApolloClient, { ApolloError } from 'apollo-client';
+import { DocumentNode } from 'graphql';
+import { ZenObservable } from 'zen-observable-ts';
+import { OperationVariables } from './types';
+
+const shallowEqual = require('fbjs/lib/shallowEqual');
+const invariant = require('invariant');
+
+export interface SubscriptionResult<TData = any> {
+  loading: boolean;
+  data?: TData;
+  error: ApolloError;
+}
+
+export interface SubscriptionProps {
+  query: DocumentNode;
+  variables?: OperationVariables;
+  children: (result: any) => React.ReactNode;
+}
+
+export interface SubscriptionState<TData = any> {
```

Comment:
Remove `export`.  I just call these `State` since they are essentially private/not exported.

Reply:
I tried this but typescript doesn't allow it with error: 

```
'extends' clause of exported class 'Subscription' has or is using private name 'SubscriptionState'.
```

Reply:
ok, if TS is going to complain and require the export (an obscure open issue in TS), then it is best to leave it as-is.